### PR TITLE
`Programming exercises`: Fix ssh local storage issue

### LINKED ISSUE #8300: `Programming Exercises`: The personal clone URL starts with undefined
### Describe the bug

After the student started the exercise, cloning was not possible as the clone URL started with undefined: `undefinedgit/IOS24INTROS6E1/ios24intros6e1-xx11xxx.git`

Reloading the page did not help. Only after opening the exercise in another browser, the correct URL for cloning was displayed.

### To Reproduce

Occurred for multiple students during the intro course, not sure how to exactly reproduce this.

### Expected behavior

The client should show a valid clone URL for all students.

### Screenshots

_No response_

### Which version of Artemis are you seeing the problem on?

6.9.5

### What browsers are you seeing the problem on?

Chrome

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by milljoniaer**
#### Client
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context
Resolves #8300 

### Description
The error occurs when SSH is not enabled, but the user still has it stored in the local storage that they prefer to use SSH when cloning the repository.
Normally, the user can select the preference in the repository clone modal, and it is saved in the local storage.
If SSH is not enabled on the server, the user no longer has the option to set the SSH option in the UI.
My suspicion:
Local CI cannot use SSH?
The users had indicated SSH as a preference when Artemis prod was still running on Atlassian.
Now SSH is disabled on the server and usessh preference still stored in local storage for some users.
=> workaround: delete local storage.

This PR ignores the `useSsh` preference, when ssh is disabled on the server.

### Steps for Testing

Prerequisites:
- 1 Student

1. Log in to Artemis
2. start programming exercise
3. Add 'jhi-usessh':"true" in your local storage (chrome can be seen in screenshot)
4. Verify that the clone url is displayed correctly and cloning into local IDE works

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### Screenshots
image

## Summary by CodeRabbit

- **Refactor**
	- Improved the logic for determining the repository URI in the clone repository button to respect both SSH settings and user preferences.

### CODE REVIEW

File: src/main/webapp/app/shared/components/clone-repo-button/clone-repo-button.component.ts [hunk 1]
```diff
@@ -104,7 +104,7 @@ export class CloneRepoButtonComponent implements OnInit, OnChanges {
     }
 
     getHttpOrSshRepositoryUri(insertPlaceholder = true): string {
-        if (this.useSsh) {
+        if (this.useSsh && this.sshEnabled) {
```
[the remark this answers is not in the document]
Comment (milljoniaer) [reply src/main/webapp/app/shared/components/clone-repo-button/clone-repo-button.component.ts:107]: this is readable

**Review by rabeatwork — APPROVED** [review]

**Review by pzdr7 — APPROVED** [review]
Code lgtm 👍

**Review by Strohgelaender — APPROVED** [review]
makes sense

**Review by rabeatwork — APPROVED** [review]
maintainer approved

**Review by Strohgelaender — APPROVED** [review]

**Pull request merged**